pymc3/stats.py

```diff
@@ -478,7 +481,7 @@ def gradient(w):
                      bounds=bounds,
                      constraints=constraints)
 
-        weights = w_fuller(w['x'])
+        weights = w_fuller(np.round(w['x'], round_to))
```

Comment:
I found the problem! I will remove the rounding trick as you suggested. Thanks for the review!